packages/playwright-core/src/server/dispatchers/elementHandlerDispatcher.ts

```diff
@@ -223,4 +224,20 @@ export class ElementHandleDispatcher extends JSHandleDispatcher implements chann
   async waitForSelector(params: channels.ElementHandleWaitForSelectorParams, metadata: CallMetadata): Promise<channels.ElementHandleWaitForSelectorResult> {
     return { element: ElementHandleDispatcher.fromNullable(this.parentScope(), await this._elementHandle.waitForSelector(metadata, params.selector, params)) };
   }
+
+  private _browserContextDispatcher(): BrowserContextDispatcher {
+    const scope = this.parentScope();
```

Comment:
Maybe a loop?

`while (scope && !(scope instanceof BrowserContextDispatcher)) scope = scope.parentScope()`

Reply:
The idea is that we keep it strongly-typed - it was wrong before.